torch/signal/windows/windows.py

```diff
@@ -0,0 +1,302 @@
+    0.4154])
+
+    >>> # Generate a symmetric cosine window.
+    >>> torch.signal.windows.cosine(10,periodic=False)
+    tensor([0.1564, 0.4540, 0.7071, 0.8910, 0.9877, 0.9877, 0.8910, 0.7071, 0.4540,
+    0.1564])
+""".format(
+        **factory_common_args
+    ),
+)
+def cosine(M: int,
+           sym: bool = True,
+           *,
+           dtype: _dtype = None,
+           layout: _layout = torch.strided,
+           device: _device = None,
+           requires_grad: bool = False) -> Tensor:
+    if dtype is None:
+        dtype = torch.get_default_dtype()
+
+    _window_function_checks('cosine', M, dtype, layout)
+
+    if M == 0:
+        return torch.empty((0,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if M == 1:
+        return torch.ones((1,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    start = 0.5
+    constant = torch.pi / (M + 1 if not sym else M)
+
+    """
+    Note that non-integer step is subject to floating point rounding errors when comparing against end;
+    thus, to avoid inconsistency, we added an epsilon equal to `step / 2` to `end`.
+    """
+    k = torch.arange(start=start * constant,
+                     end=(start + (M - 1)) * constant + constant / 2,
+                     step=constant,
+                     dtype=dtype,
+                     layout=layout,
+                     device=device,
+                     requires_grad=requires_grad)
+
+    return torch.sin(k)
+
+
+@_add_docstr(
+    r"""
+Computes a window with a gaussian waveform.
+
+The gaussian window is defined as follows:
+
+.. math::
+    w(n) = \exp{\left(-\left(\frac{n}{2\sigma}\right)^2\right)}
+    """,
+    r"""
+
+Args:
+    M (int): the length of the output window.
+        In other words, the number of points of the cosine window.
+    std (float): the standard deviation of the gaussian. It controls how narrow or wide the window is.
+    sym (bool, optional): If `False`, returns a periodic window suitable for use in spectral analysis.
+        If `True`, returns a symmetric window suitable for use in filter design. Default: `True`
+
+.. note::
+    The window is normalized to 1 (maximum value is 1), however, the 1 doesn't appear if `M` is even
+    and `sym` is `True`.
+
+Keyword args:
+    {dtype}
+    {layout}
+    {device}
+    {requires_grad}
+
+Examples:
+    >>> # Generate a gaussian window without keyword args.
+    >>> torch.signal.windows.gaussian(10)
+    tensor([1.9287e-22, 1.2664e-14, 1.5230e-08, 3.3546e-04, 1.3534e-01, 1.0000e+00,
+    1.3534e-01, 3.3546e-04, 1.5230e-08, 1.2664e-14])
+
+    >>> # Generate a symmetric gaussian window and standard deviation equal to 0.9.
+    >>> torch.signal.windows.gaussian(10,periodic=False,std=0.9)
+    tensor([3.7267e-06, 5.1998e-04, 2.1110e-02, 2.4935e-01, 8.5700e-01, 8.5700e-01,
+    2.4935e-01, 2.1110e-02, 5.1998e-04, 3.7267e-06])
+""".format(
+        **factory_common_args
+    ),
+)
+def gaussian(M: int,
```

Comment:
Updated signature.

Reference: https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.windows.gaussian.html

I find it very annoying that Scipy uses `std` as a required arg rather than a keyword argument as it does with `tau` for the exponential window.

Reply:
I'm happy making it kwarg-only with default 1. It's good to follow SciPy, but we can always improve on it :)

Reply:
Done.